podpac/core/node.py

```diff
@@ -1208,33 +1208,97 @@ def get_ui_spec(cls, help_as_html=False):
         return spec
 
 
-def _lookup_input(nodes, name, value):
+def _lookup_input(nodes, name, value, definition):
     # containers
     if isinstance(value, list):
-        return [_lookup_input(nodes, name, elem) for elem in value]
+        return [_lookup_input(nodes, name, elem, definition) for elem in value]
 
     if isinstance(value, dict):
-        return {k: _lookup_input(nodes, name, v) for k, v in value.items()}
+        return {k: _lookup_input(nodes, name, v, definition) for k, v in value.items()}
 
     # node reference
     if not isinstance(value, six.string_types):
         raise ValueError(
             "Invalid definition for node '%s': invalid reference '%s' of type '%s' in inputs"
             % (name, value, type(value))
         )
+    # node not yet discovered yet
+    if not value in nodes:
+        # Look for it in the definition items:
+        for found_name, d in definition.items():
+            if value != found_name:
+                continue
+
+             # get node class
+            module_root = d.get("plugin", "podpac")
```

Comment:
Is the code below here copied and pasted from somewhere else?

Reply:
Yes. The code from `from_definition` worked here as well, with a few modifications. Happy to make a function to encapsulate this loop and call it in both locations.?

Reply:
Attempted to fix this with commit 3fb5048ef08d00fd69e95d05343ebe02db73fc98